girdermedviewer/app/girder/utils.py

```diff
@@ -42,18 +45,31 @@ def __del__(self):
     def get_item_files(self, item):
         return self.girder_client.listFile(item["_id"])
 
-    @contextmanager
     def download_file(self, file):
-        """
-        :param forced_cache can be used to override the cache mode for a specific file
-        """
-        file_path = os.path.join(self.temp_dir_path, file["_id"], file["name"])
+        file_path = os.path.join(self.temp_dir_path, file['_id'], file["name"])
         if not os.path.exists(file_path):
             logger.info(f"Download {file['name']} to {file_path}")
             self.girder_client.downloadFile(
                 file["_id"],
                 file_path
             )
+        return file_path

```

Comment:
I now think that the caching handling should be in `fetch_file` and not in `download_file` anymore.

Let's make `download_file` an internal function -> `_download_file(self, file, file_path)` and handle the caching (i.e. `if not os.path.exists(file_path)`) in `fetch_file()`.